File: israeli-id-validation.go
package main

import (
    "fmt"
    "strconv"
)

func isValidIsraeliID(id string) bool {
    // Add leading zeroes if ID is less than 9 digits
    id = fmt.Sprintf("%09s", id)

    // Convert ID to array of integers
    idArr := make([]int, 9)
    for i := 0; i < 9; i++ {
        idArr[i], _ = strconv.Atoi(string(id[i]))
    }

    // Multiply first 8 digits by check array
    checkArr := [8]int{1, 2, 1, 2, 1, 2, 1, 2}
    var calcArr [8]int
    for i := 0; i < 8; i++ {
        calcArr[i] = idArr[i] * checkArr[i]
    }

    // Calculate sum of resulting array
    var sum int
    for _, val := range calcArr {
        if val > 9 {
            sum += val - 9
        } else {
            sum += val
        }
    }

    // Calculate check digit
    checkDigit := (10 - (sum % 10)) % 10

    // Compare check digit with last digit of ID
    return idArr[8] == checkDigit
}

File: israeli-id-validation.py
def is_valid_israeli_id(id: str) -> bool:
    # Add leading zeroes if ID is less than 9 digits
    id = id.zfill(9)

    # Convert ID to array of integers
    id_arr = [int(i) for i in id]

    # Multiply first 8 digits by check array
    check_arr = [1, 2, 1, 2, 1, 2, 1, 2]
    calc_arr = [id_arr[i] * check_arr[i] for i in range(8)]

    # Calculate sum of resulting array
    sum = sum(val - 9 if val > 9 else val for val in calc_arr)

    # Calculate check digit
    check_digit = (10 - (sum % 10)) % 10

    # Compare check digit with last digit of ID
    return id_arr[8] == check_digit

